### ISSUE TITLE
Encoding directly into `io::Write` and `fmt::Write`

### ISSUE THREAD

**Issue by tyranron** [fbernier/base62#16]
It would be nice to have the following functions in this crate:
```rust
fn encode_fmt<T: Into<u128>, W: fmt::Write + ?Sized>(num: T, f: &mut W) -> fmt::Result {}
fn encode_io<T: Into<u128>, W: io::Write + ?Sized>(num: T, w: &mut W) -> io::Result<()> {}
```

This will allow stripping redundant allocations when implementing Base62-encoded `Display`, for example.

**Comment by tyranron** [comment]
@fbernier wow! Many thanks! Could you cut a new release with these changes?

**Comment by fbernier** [comment]
> @fbernier wow! Many thanks! Could you cut a new release with these changes?

yes, sorry for the silence. That was a great suggestion and I'll be cutting a release today.

Thanks!

**Comment by fbernier** [comment]
published as 2.1.0
